app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,348 @@
+ */
+@Composable
+fun SyncedTabsErrorItem(errorText: String, errorButton: SyncedTabsListItem.ErrorButton? = null) {
+    Box(
+        Modifier
+            .padding(all = 16.dp)
+            .height(IntrinsicSize.Min)
+    ) {
+        val dashColor = FirefoxTheme.colors.borderDefault
+
+        Canvas(Modifier.fillMaxSize()) {
+            drawRoundRect(
+                color = dashColor,
+                style = Stroke(
+                    width = 2.dp.toPx(),
+                    pathEffect = PathEffect.dashPathEffect(floatArrayOf(4.dp.toPx(), 4.dp.toPx()), 0f)
+                ),
+                cornerRadius = CornerRadius(
+                    x = 8.dp.toPx(),
+                    y = 8.dp.toPx()
+                ),
+            )
+        }
+
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            PrimaryText(
+                text = errorText,
+                modifier = Modifier.fillMaxWidth(),
+                fontSize = 14.sp
+            )
+
+            errorButton?.let { SyncedTabsErrorButton(buttonText = it.buttonText, onClick = it.onClick) }
+        }
+    }
+}
+
+/**
+ * Error button UI within SyncedTabsErrorItem
+ *
+ * @param buttonText The error button's text and accessibility hint.
+ * @param onClick The lambda called when the button is clicked.
+ */
+@Composable
+fun SyncedTabsErrorButton(buttonText: String, onClick: () -> Unit) {
+    Button(
+        onClick = onClick,
+        modifier = Modifier
+            .clip(RoundedCornerShape(size = 4.dp))
+            .padding(top = 8.dp),
+        elevation = ButtonDefaults.elevation(defaultElevation = 0.dp, pressedElevation = 0.dp),
+        colors = ButtonDefaults.outlinedButtonColors(backgroundColor = FirefoxTheme.colors.actionPrimary),
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.ic_sign_in),
+            contentDescription = null,
+            tint = FirefoxTheme.colors.textInverted,
+        )
+
+        Spacer(Modifier.width(8.dp))
+
+        Text(
+            text = buttonText,
+            modifier = Modifier.align(Alignment.CenterVertically),
+            color = FirefoxTheme.colors.textInverted,
+            fontSize = 14.sp,
+            fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+            maxLines = 2
+        )
+    }
+}
+
+/**
+ * UI to be displayed when a user's device has no synced tabs.
+ */
+@Composable
+fun SyncedTabsNoTabsItem() {
+    SecondaryText(
+        text = stringResource(R.string.synced_tabs_no_open_tabs),
+        modifier = Modifier
+            .padding(horizontal = 16.dp, vertical = 8.dp)
+            .fillMaxWidth(),
+        fontSize = 16.sp,
+        maxLines = 1
+    )
+}
+
+@Composable
+@Preview
+private fun SyncedTabsListItemsPreview() {
+    FirefoxTheme(darkTheme = true) {
+        Column(Modifier.background(FirefoxTheme.colors.layer2)) {
```

Comment:
These should be layer1. See https://www.figma.com/file/YcQC7rU1SQJx49kIAXZhcI/Component-and-Color-Usage-Examples?node-id=78%3A11241